tests/template_tests/test_loaders.py

```diff
@@ -191,11 +191,11 @@ def test_permissions_error(self):
             tmppath = os.path.join(tmpdir, tmpfile.name)
             os.chmod(tmppath, 0o0222)
             with self.set_dirs([tmpdir]):
-                with self.assertRaisesMessage(IOError, 'Permission denied'):
+                with self.assertRaisesMessage(PermissionError, 'Permission denied'):
                     self.engine.get_template(tmpfile.name)
 
     def test_notafile_error(self):
-        with self.assertRaises(IOError):
+        with self.assertRaises(IsADirectoryError):
```

Comment:
This isn't working on Windows:
```
======================================================================
ERROR: test_notafile_error (template_tests.test_loaders.FileSystemLoaderTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Users\tim\code\django\tests\template_tests\test_loaders.py", line 199, in test_notafile_error
    self.engine.get_template('first')
  File "c:\users\tim\code\django\django\template\engine.py", line 143, in get_template
    template, origin = self.find_template(template_name)
  File "c:\users\tim\code\django\django\template\engine.py", line 125, in find_template
    template = loader.get_template(name, skip=skip)
  File "c:\users\tim\code\django\django\template\loaders\base.py", line 24, in get_template
    contents = self.get_contents(origin)
  File "c:\users\tim\code\django\django\template\loaders\filesystem.py", line 23, in get_contents
    with open(origin.name, encoding=self.engine.file_charset) as fp:
PermissionError: [Errno 13] Permission denied: 'C:\\Users\\tim\\code\\django\\tests\\template_tests\\templates\\first'
```
It seems that `with self.assertRaises(IOError):` also catches `PermissionError`.

Should we go with `PermissionError if sys.platform.startswith('win') else IsADirectoryError` here?

Reply:
That works for me. Would you like me to open a PR with that fix?

I'd also be OK catching the parent `OSError` if you think the platform specificity isn't useful in this case.